File: fetchChicagoTime.js
export async function fetchChicagoTime() {
  try {
    const res = await fetch('https://worldtimeapi.org/api/timezone/America/Chicago');
    const json = await res.json();
    if (json.datetime) {
      return new Date(json.datetime).toLocaleString('en-US', { timeZone: 'America/Chicago' });
    }
    return '';
  } catch (e) {
    return 'Error fetching time';
  }
}

File: src/main/kotlin/FetchChicagoTime.kt
import kotlinx.coroutines.await
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.Json
import kotlin.js.Date

@Serializable
data class TimeResponse(val datetime: String? = null)

suspend fun fetchChicagoTime(): String {
    return try {
        val response = fetch("https://worldtimeapi.org/api/timezone/America/Chicago").await()
        val jsonText = response.text().await()
        val timeData = Json { ignoreUnknownKeys = true }.decodeFromString<TimeResponse>(jsonText)
        
        timeData.datetime?.let { datetime ->
            Date(datetime).toLocaleString("en-US", js("{ timeZone: 'America/Chicago' }"))
        } ?: ""
    } catch (e: Exception) {
        "Error fetching time"
    }
}